letta/functions/schema_generator.py

```diff
@@ -38,15 +38,29 @@ def type_to_json_schema_type(py_type):
 
     # Mapping of Python types to JSON schema types
     type_map = {
+        # Basic types
         int: "integer",
         str: "string",
         bool: "boolean",
         float: "number",
-        list[str]: "array",
-        # Add more mappings as needed
+        # Collections
+        List[str]: "array",
+        List[int]: "array",
+        list: "array",
+        tuple: "array",
+        set: "array",
+        # Dictionaries
+        dict: "object",
+        Dict[str, Any]: "object",
+        # Special types
+        None: "null",
+        type(None): "null",
+        # Optional types
+        Optional[str]: "string",
```

Comment:
@mattzh72 maybe it makes sense to kill this?
`Optional[str]: "string",`

because of the code above? `if is_optional(py_type):`

Reply:
Hm yeah agreed - seems like above catches all the Optionals and unwraps them.